src/kv/tx.h

```diff
@@ -2,6 +2,22 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
+// The transaction engine supports producing a transaction dependencies. This is
+// materialized by providing a sequence number after which the current
+// transaction must be run. The current use case for dependency tracking is to
+// enable parallel execution of transactions on the backup, and as such we only
+// track dependencies when running BFT consensus protocol.
```

Comment:
I must be missing something, but this sounds incomplete. This gives a lower bound ("this transaction must read at at-least this version, which is the version _I_ read it at"), but to ensure all executions read the same value we also need an upper bound - this transaction must not execute too late or your value may have changed by the time you execute this. How is that dependency tracked and enforced?

Reply:
Ill add it to the comment but  transaction execution is started in the total order